x-pack/plugins/integration_assistant/server/graphs/log_type_detection/graph.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  ActionsClientChatOpenAI,
+  ActionsClientSimpleChatModel,
+} from '@kbn/langchain/server/language_models';
+import type { StateGraphArgs } from '@langchain/langgraph';
+import { END, START, StateGraph } from '@langchain/langgraph';
+import type { LogFormatDetectionState } from '../../types';
+import { EX_ANSWER_LOG_TYPE } from './constants';
+import { handleLogFormatDetection } from './detection';
+import { SamplesFormat } from '../../../common';
+
+const graphState: StateGraphArgs<LogFormatDetectionState>['channels'] = {
+  lastExecutedChain: {
+    value: (x: string, y?: string) => y ?? x,
+    default: () => '',
+  },
+  rawSamples: {
+    value: (x: string[], y?: string[]) => y ?? x,
+    default: () => [],
+  },
+  exAnswer: {
+    value: (x: string, y?: string) => y ?? x,
+    default: () => '',
+  },
+  finalized: {
+    value: (x: boolean, y?: boolean) => y ?? x,
+    default: () => false,
+  },
+  samplesFormat: {
+    value: (x: SamplesFormat, y?: SamplesFormat) => y ?? x,
+    default: () => ({ name: 'json' }),
```

Comment:
I am a bit unsure why this has a default value? I assume the invoke of the graph will never have samplesFormat set, and its only set after the format has been detected? If that's the case then this should also be empty.

Reply:
The `samplesFormat` is of type [SamplesFormat](https://github.com/elastic/kibana/pull/190407/files#diff-b115a6be8ec9464eb85b15041853d0057244eecc6db0c3f82b6b1eb7b2bdf0c7R73-R74) which needs a `name` as a default object [ which takes only a specific set of values from the [enum](https://github.com/elastic/kibana/pull/190407/files#diff-b115a6be8ec9464eb85b15041853d0057244eecc6db0c3f82b6b1eb7b2bdf0c7R58-R65) ]

Reply:
Yeah.. `unsupported` is a good start for default. Since `json` format does not enter this graph.

Not sure why nullable? Since this is an object with `name` as mandatory parameter.